File: IOS/Sources/OpenRoaming/OpenRoaming/Services/GetPrivacySettingsService.swift
//
//  GetPrivacySettingsService.swift
//  OpenRoaming
//
//  Created by Fabiana Garcia on 5/18/20.
//  Copyright © 2020 Cisco. All rights reserved.
//

import Foundation

class GetPrivacySettingsService: Service {
    
    override func onStart(headerParams: Array<HeaderParam>, serviceHandler: @escaping ServiceHandler) {
        let urlString = Urls.privacySettingsUrl()
        let requestData = RequestData(url: urlString, headerParams: headerParams, method: HTTPMethod.GET)
        call(requestData: requestData, serviceHandler: serviceHandler)
    }
}

File: Android/src/main/java/com/cisco/or/sdk/services/GetPrivacySettingsService.kt
package com.cisco.or.sdk.services

import android.content.Context
import com.cisco.or.sdk.types.ServiceHandler
import com.cisco.or.sdk.utils.Urls

internal object GetPrivacySettingsService : Service(){

    override fun onStart(context: Context, headerParams: ArrayList<HeaderParam>, serviceHandler: ServiceHandler) {
        val url = Urls.privacySettingsUrl()
        val requestData = RequestData(url, headerParams, HTTP_METHOD.GET, null)
        call(context, requestData, RESPONSE_FORMATTER.TEXT, serviceHandler)
    }
}